### PR TITLE
More enhancements to demos.

### PR DESCRIPTION
**Pull request by merriam**
Added some more functionality to the kivy-catalog.   I hope to slowly improve this demo until it shows much of the framework.   Minor fixes to the showcase demo.

### CODE REVIEW

**Comment by merriam** [comment]
GitHub is being a bit unclear with this request.   There were three requests  a moment ago:

Added some more functionality to the kivy-catalog. I hope to slowly improve this demo until it shows much of the framework. Minor fixes to the showcase demo.

Label's parameter 'padding' was incorrectly listed in the docs.

Fixes to documentation and error messages in 'atlas.py'

Hopefully, the combined pull request won't create a problem.   A total of 5 files changed.

File: examples/demo/showcase/main.py [hunk 1]
```diff
@@ -75,8 +75,8 @@ def go_screen(self, idx):
         self.root.ids.sm.switch_to(self.load_screen(idx), direction='left')
         self.update_sourcecode()
 
-    def go_higherarchy_previous(self):
-        ahr = self.higherarchy
+    def go_heirarchy_previous(self):
+        ahr = self.heirarchy
```
Comment (dessant) [inline examples/demo/showcase/main.py]: Lines 30, 78, 79, the third revision will be the best: hierarchy.

File: kivy/atlas.py [hunk 2]
```diff
@@ -4,18 +4,21 @@
...
 .. versionadded:: 1.1.0
 
-Atlas is a class for managing texture atlases: packing multiple textures into
+Atlas manages texture atlases: packing multiple textures into
 one. With it, you reduce the number of images loaded and speedup the
 application loading.
 
-An Atlas is composed of:
+This file contains both the Atlas class and command line processing for creating
+an atlas from a set of individual PNG files.   The command line section requires
```
Comment (dessant) [inline kivy/atlas.py]: That space between sentences could be removed.

File: examples/demo/showcase/main.py (hunk 1 above)
Comment (merriam) [inline examples/demo/showcase/main.py]: I agree.  I think 30, 78, and 79 were in a single commit.  Just merge that?

File: kivy/atlas.py [hunk 3]
```diff
@@ -64,14 +72,15 @@
...
+    $ python -m kivy.atlas myatlas 256x256 *.png
     Atlas created at myatlas.atlas
-    1 image have been created
+    1 image has been created
     $ ls
     bubble.png bubble-red.png button.png button-down.png myatlas.atlas
     myatlas-0.png
 
-As you can see, we get 2 new files: ``myatlas.atlas`` and ``myatlas-0.png``.
+As you can see, we get 2 new files: ``myatlas.atlas`` and ``myatlas-0.png``.  ``myatlas-0.png``
```
Comment (dessant) [inline kivy/atlas.py]: Two spaces between sentences.

**Comment by dessant** [comment]
@merriam besides very minor typos, a huge +1 for this. Thanks for your contributions so far.

**Comment by dessant** [comment]
@merriam, you could make those small fixes in a new commit for your own fork, master branch, the pr is automatically updated and we can merge them together.

File: examples/demo/showcase/main.py (hunk 1 above)
Comment (dessant) [inline examples/demo/showcase/main.py]: I mean the correct form is "hierarchy", we could merge it after the fix.

Comment (merriam) [inline examples/demo/showcase/main.py]: My spelling fails as well!  :)

Comment (merriam) [inline examples/demo/showcase/main.py]: This should be ready to merge now.
1.  Fixed the hierarchy misspellings.
2.  Fixed the extra line.
3.  Fixed the long-line style warnings from 'make style'.

I solemnly swear to make the next set of commits on a branch other than
master. :).    Next set of commits will be either some tests or fixing the
html doc generation.
https://groups.google.com/forum/#!topic/kivy-dev/4_DzJUuZAXM

Charles



On Mon, Jan 26, 2015 at 12:45 PM, Charles Merriam <charles.merriam@gmail.com

**Comment by dessant** [comment]
@merriam as you noted in your commit message, the best would be to work on branches. You can always start over from master if you mess up. This pr looks broken in its current form.

**Comment by dessant** [comment]
Sorry, just got your mail with the diff comment, it does not seem to be visible here.
The problem is that there are some wrong commits here, like this https://github.com/merriam/kivy/commit/409fd642ff25bd047061fa72f9a50b26d9afbf42. As you see you pr has grown to 2k lines.

My suggestion is to port over the useful commits to a new branch somehow, and do a new, clean pr.

**Comment by dessant** [comment]
Btw it is perfectly ok to make these mistakes, we do them too. We regard your contributions very useful, so branch up ;) the good work.

**Comment by dessant** [comment]
Closing as it was resubmitted:
https://github.com/kivy/kivy/pull/2924
https://github.com/kivy/kivy/pull/2925
https://github.com/kivy/kivy/pull/2926

**Pull request closed without merging**